src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js

```diff
@@ -97,42 +97,14 @@ define(function() {
                 setData.call(this, {ids: ids}, false);
             }, this);
 
-            // add image to the selected images grid
-            this.sandbox.on(
-                'sulu.media-selection-overlay.' + this.options.instanceName + '.record-selected',
-                function(itemId, item) {
-                    var data = this.getData(),
-                        index = data.ids.indexOf(itemId);
-
-                    if (index > -1) {
-                        return;
-                    }
-
-                    data.ids.push(itemId);
-                    this.setData(data, false);
-                    this.addItem(item);
-                }.bind(this)
-            );
-
-            // remove image to the selected images grid
-            this.sandbox.on(
-                'sulu.media-selection-overlay.' + this.options.instanceName + '.record-deselected',
-                function(itemId) {
-                    var data = this.getData(),
-                        index = data.ids.indexOf(itemId);
-
-                    if (index > -1) {
-                        data.ids.splice(index, 1);
-                    }
-
-                    this.setData(data, false);
-                    this.removeItemById(itemId);
-                }.bind(this)
-            );
-
             this.sandbox.on('sulu.media-selection.' + this.options.instanceName + '.add-button-clicked', function() {
+                var items = _.map(this.getData().ids, function(id) {
+                    return {id: id};
```

Comment:
That could be changed, but keep in mind that it is at the interface of an event-handler, that would be a BC break.